woocommerce/woocommerce-ios
# Edit Product: fetch product shipping class and update shipping settings remotely

Fixes #1422 

## Summary

This PR completes the core parts of Product Shipping Settings by fetching the product's shipping class on Edit Shipping Settings and saving the settings remotely

## Changes

- In `Product` encoding function, encoded the properties in shipping settings
- Updated `ProductDimensions` readonly model to implement `Codable` instead of just `Decodable` to enable default encoding
- Updated `PaginatedProductShippingClassListSelectorDataSource` initializer to take in the selected shipping class
- In `ProductFormViewController`, handled the completion block of `ProductShippingSettingsViewController`
- In `ProductShippingSettingsViewController`, added a completion handler in its initializer
- In `ProductShippingSettingsViewController`, added navigation to a list selector for product's shipping class
- In `ProductUpdater`, added a function to update the shipping settings
- Updated/added unit tests

## Testing

Prerequisite: the store has at least two shipping classes
- Launch the app
- Go to the Products tab
- Tap on a Product
- Tap "Edit" in the navigation bar
- Tap the shipping settings row to update the shipping settings and navigate back to the Edit Product page without tapping "Done" --> the shipping settings shouldn't be saved after tapping the shipping settings row again
- Update the shipping settings with a shipping class, then tap "Done"
- Tap "Done" on the Edit Product page --> it should go back to the Product Details page with the shipping settings updated
- On web, also check the shipping settings are updated correctly
- Back in the app, tap "Edit" and then tap the shipping settings row to update the shipping settings to a different shipping class
- Tap "Done" on the shipping settings screen
- Tap "Done" on the Edit Product page --> it should go back to the Product Details page with the shipping settings updated
- On web, also check the shipping settings are updated correctly

## Example screenshots

Edit Shipping Settings | Shipping class picker
-- | --
![Simulator Screen Shot - iPhone 8 - 2019-12-17 at 16 25 58](https://user-images.githubusercontent.com/1945542/70978491-d0753600-20ea-11ea-9c60-59876e8ec683.png) | ![Simulator Screen Shot - iPhone 8 - 2019-12-17 at 16 26 00](https://user-images.githubusercontent.com/1945542/70978492-d0753600-20ea-11ea-9761-70d5389424aa.png)


Update release notes:
